modules/auxiliary/scanner/misc/cisco_smart_install.rb

```diff
@@ -57,10 +65,79 @@ def smi?
     end
   end
 
-  def run_host(_ip)
+  def start_tftp
+    print_status("Starting TFTP Server...")
+    @tftp = Rex::Proto::TFTP::Server.new(69, '0.0.0.0', { 'Msf' => framework, 'MsfExploit' => self })
+    @tftp.incoming_file_hook = Proc.new{|info| process_incoming(info) }
+    @tftp.start
+    add_socket(@tftp.sock)
+    @main_thread = ::Thread.current
+  end
+
+  def cleanup
+    # Cleanup is called once for every single thread
+    if ::Thread.current == @main_thread
+      # Wait 5 seconds for background transfers to complete
+      print_status("Providing some time for transfers to complete...")
+      ::IO.select(nil, nil, nil, 5.0)
+
+      print_status("Shutting down the TFTP service...")
```

Comment:
move inside the `if` so that it only prints that it is shutting down the service when there is actually a service to shut down.